plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1426,6 +1433,10 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(
             transactionLogWriter.flush();
             writeCommitted = true;
 
+            if (handle.replace() && readVersion < commitVersion) {
```

Comment:
why do we need the check `readVersion < commitVersion` ?
when doing replace this should always be true right?

Reply:
Linking to [this previous comment](https://github.com/trinodb/trino/pull/21609#discussion_r1577900914): on the first statement of the test `TestDeltaLakeFileOperations#testCreateOrReplaceTableAsSelect` when we do `CREATE OR REPLACE TABLE test_create_or_replace_as_select AS SELECT 1 col_name` we have `handle.replace()` as true, even though it's a new table:
<img width="764" alt="Screenshot 2024-04-25 at 13 08 31" src="https://github.com/trinodb/trino/assets/15678584/71fea9e1-046b-4963-b35b-ab5aa463869b">
Checking on the versions makes it possible to not trigger a checkpoint write unnecessarily.

Reply:
i see now what you mean.
we should then have a more easy to read solution then.

```suggestion
if (handle.replace() && handle.readVersion().isEmpty())
```

Reply:
Done